File: examples/hello.cpp
// Cross-compiled C++ using the SDK's libc++.
//
//   source ../xmac-sdk/env.sh
//   $CXX -o hello-cpp hello.cpp
#include <iostream>
#include <numeric>
#include <string>
#include <vector>

int main() {
  std::vector<std::string> v{"hello", "from", "libc++", "on", "macOS"};
  std::cout << std::accumulate(v.begin(), v.end(), std::string{},
                               [](auto a, auto b) {
                                 return a.empty() ? b : a + " " + b;
                               })
            << "\n";
}

File: examples/hello.m
// Cross-compiled Objective-C with ARC.
//
//   source ../xmac-sdk/env.sh
//   $CC -fobjc-arc -framework Foundation -o hello-objc hello.m
#import <Foundation/Foundation.h>

int main(void) {
  @autoreleasepool {
    NSLog(@"%@", [NSString stringWithFormat:@"objc on %@", @"macOS"]);
  }
  return 0;
}
